### DISCUSSION TITLE
Where it is `CookieHelper` (Keycloak 25.0.+)?

### DISCUSSION THREAD

**Discussion by jairorrfb** [keycloak/keycloak#32584] [Q&A] [answered]
I am trying to create a custom authenticator following the [official guide here](https://www.keycloak.org/docs/25.0.0/server_development/#_auth_spi), but I can't find the `addCookie()` method from `CookieHelper`.

It looks like [Keycloak 23](https://www.keycloak.org/docs-api/23.0.0/javadocs/org/keycloak/services/util/CookieHelper.html) was the last version which this helper was shipped in, and the current Keycloak 25 documentation is out-of-date.

Does anybody know how to help me?

**Comment by danielFesenmeyer** [comment, marked as answer]
It has been replaced with CookieSpi / CookieProvider: [CookieProvider.java
](https://github.com/keycloak/keycloak/blob/main/server-spi-private/src/main/java/org/keycloak/cookie/CookieProvider.java)
